src/lib/operators/jit_operator/jit_operations.hpp

```diff
@@ -78,28 +86,83 @@ const auto jit_power = [](const auto a, const auto b) -> decltype(std::pow(a, b)
 
 /* Aggregate operations */
 const auto jit_increment = [](const auto a, const auto b) -> decltype(b + 1) { return b + 1; };
-const auto jit_maximum = [](const auto a, const auto b) { return std::max(a, b); };
-const auto jit_minimum = [](const auto a, const auto b) { return std::min(a, b); };
+struct JitMaximum {
+  OPTNONE std::string operator()(const std::string& a, const std::string& b) const { return std::max(a, b); }
+  template <typename T1, typename T2,
+            typename = typename std::enable_if_t<std::is_scalar_v<T1> && std::is_scalar_v<T2>>>
+  auto operator()(const T1 a, const T2 b) const {
+    return std::max(a, b);
+  }
+};
+const auto jit_maximum = JitMaximum();
+struct JitMinimum {
+  OPTNONE std::string operator()(const std::string& a, const std::string& b) const { return std::min(a, b); }
```

Comment:
Does this mean that we have to write `OPTNONE` in any place where  we might have to handle a string?

Reply:
Yes, but I tried to always use a separate string function implementation so that the functions are still optimized for non-string types. Currently, only the `get` and `set` methods in `JitVariantVector` and `JitHashmapValue` are never optimized.